File: app/src/main/java/com/savvasdalkitsis/gameframe/draw/view/SimplePalettesAdapter.java
package com.savvasdalkitsis.gameframe.draw.view;

import android.support.v7.widget.RecyclerView;
import android.view.ViewGroup;

import com.savvasdalkitsis.gameframe.draw.model.Palette;


class SimplePalettesAdapter extends RecyclerView.Adapter<PaletteViewHolder> {

    private Palette[] palettes;
    private AddNewPaletteSelectedListener onAddNewPaletteSelectedListener;

    SimplePalettesAdapter(Palette[] palettes) {
        this.palettes = palettes;
    }

    @Override
    public PaletteViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        return new PaletteViewHolder(parent, false);
    }

    @Override
    public void onBindViewHolder(PaletteViewHolder holder, int position) {
        holder.clearListeners();
        holder.bind(palettes[position]);
        holder.setOnClickListener(v -> onAddNewPaletteSelectedListener.onAddNewPalletSelected(palettes[position]));
    }

    @Override
    public int getItemCount() {
        return palettes.length;
    }

    void setOnAddNewPaletteSelectedListener(AddNewPaletteSelectedListener onAddNewPaletteSelectedListener) {
        this.onAddNewPaletteSelectedListener = onAddNewPaletteSelectedListener;
    }

}

File: app/src/main/java/com/savvasdalkitsis/gameframe/feature/draw/view/SimplePalettesAdapter.kt
package com.savvasdalkitsis.gameframe.feature.draw.view

import android.support.v7.widget.RecyclerView
import android.view.ViewGroup

import com.savvasdalkitsis.gameframe.feature.draw.model.Palette


internal class SimplePalettesAdapter(private val palettes: Array<Palette>) : RecyclerView.Adapter<PaletteViewHolder>() {

    private var onAddNewPaletteSelectedListener: AddNewPaletteSelectedListener? = null

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int) =
            PaletteViewHolder(parent, false)

    override fun onBindViewHolder(holder: PaletteViewHolder, position: Int) = with(holder) {
        clearListeners()
        bind(palettes[position])
        setOnClickListener { onAddNewPaletteSelectedListener?.onAddNewPalletSelected(palettes[position]) }
    }

    override fun getItemCount() = palettes.size

    fun setOnAddNewPaletteSelectedListener(onAddNewPaletteSelectedListener: AddNewPaletteSelectedListener) {
        this.onAddNewPaletteSelectedListener = onAddNewPaletteSelectedListener
    }

}
